src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/BaseRunner.cs

```diff
@@ -42,7 +47,7 @@ public async Task<(TestResult, string)> RunAsync(TestCase testCase)
                 (result, message) = RunAsync2(testCase).Result;
             });
             t.Start();
-            var success = t.Join(Timeout);
+            var success = t.Join(System.Diagnostics.Debugger.IsAttached ? TimeSpan.FromSeconds(20000) : Timeout);
```

Comment:
>System.Diagnostics.Debugger.IsAttached ? TimeSpan.FromSeconds(20000) [](http://example.com/codeflow?start=33&length=68)

The looping is really the proper way to do this. 
With this approach, this always disabled the timeout if they run under debugger (even if not hitting breakpoints).

Reply:
Pushed new changes